app/controllers/locations_controller.rb

```diff
@@ -1,14 +1,13 @@
 class LocationsController < ApplicationController
   def index
-
+    @locations = Location.all
   end
 
   def show
-    @location = Location.find(params['id']).name
-    # params[location] will be passed into items_by_location and
-    # only items within a certain location will be displayed
-    city_group = Item.items_by_location(@location)
-    @items = city_group[@location]
+    @location = Location.find(params[:id])
+    location_name = @location[:name]
+    city_group = Item.items_by_location(@location[:name])
+    @location_items = city_group[location_name]
   end
 
 end
```

Comment:
A Location has many items. Couldn't you just call @location.items in the view? That way all you need is the one line here @location = Location.find(params[:id]).